ObjCVideoQuickstart/ViewController.m

```diff
@@ -0,0 +1,314 @@
+        
+        UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self
+                                                                              action:@selector(flipCamera)];
+        [self.previewView addGestureRecognizer:tap];
+    }
+}
+
+- (void)flipCamera {
+    [self.camera flipCamera];
+}
+
+- (void)prepareLocalMedia {
+    
+    // We will offer local audio and video when we connect to room.
+    
+    // Adding local audio track to localMedia
+    self.localAudioTrack = [self.localMedia addAudioTrack:YES];
+    
+    // Adding local video track to localMedia and starting local preview if it is not already started.
+    if (self.localMedia.videoTracks.count == 0) {
+        [self startPreview];
+    }
+}
+
+- (IBAction)connectButtonPressed:(id)sender {
+    if([self.accessToken isEqualToString:@"TWILIO_ACCESS_TOKEN"]) {
+        [self logMessage:@"Please provide a valid token to connect to a room."];
+        return;
+    }
+    
+    // Creating a video client with the use of the access token.
+    if (!self.client) {
+        self.client = [TVIVideoClient clientWithToken:self.accessToken];
+        if (!self.client) {
+            [self logMessage:@"Failed to create video client"];
+            return;
+        }
+    }
+    
+    // Preparing local media to offer in when we connect to room.
+    [self prepareLocalMedia];
+    
+    TVIConnectOptions *connectOptions = [TVIConnectOptions optionsWithBlock:^(TVIConnectOptionsBuilder * _Nonnull builder) {
+        
+        // We will set the prepared local media in connect options.
+        builder.localMedia = self.localMedia;
+        
+        // The name of the Room where the Client will attempt to connect to. Please note that if you pass an empty
+        // Room `name`, the Client will create one for you. You can get the name or sid from any connected Room.
+        builder.name = self.roomTextField.text;
+    }];
+    
+    // Attempting to connect to room with connect options.
+    self.room = [self.client connectWithOptions:connectOptions delegate:self];
+    
+    [self logMessage:[NSString stringWithFormat:@"Attempting to connect to room %@", self.roomTextField.text]];
+    
+    [self toggleView];
+    [self dismissKeyBoard];
+}
+
+- (IBAction)disconnectButtonPressed:(id)sender {
+    [self.room disconnect];
+}
+
+- (IBAction)micButtonPressed:(id)sender {
+    // We will toggle the mic to mute/unmute and change the title according to the user action. 
+    
+    if ([self.localMedia.audioTracks count] > 0) {
+        self.localMedia.audioTracks[0].enabled = !self.localMedia.audioTracks[0].isEnabled;
+        
+        //toggle the button title
+        if (self.localMedia.audioTracks[0].isEnabled) {
+            [self.micButton setTitle:@"Mute" forState:UIControlStateNormal];
+        } else {
+            [self.micButton setTitle:@"Unmute" forState:UIControlStateNormal];
+
+        }
+    }
+}
```

Comment:
Food for thought, we may want to consider adding back convenience APIs like JS has. It feels a little clunky when you don't hold on to the audio track.